test/Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests/Common/KafkaE2EFixture.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Logging;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.Common
+{
+	/* Common fixture for all language test case classes which does -
+	* Azure Infra setup and Func Apps Startup
+	* Stopping Func Apps and Azure Infra cleanup
+	*/
+	public class KafkaE2EFixture : IAsyncLifetime
+	{
+		private Language _language;
+		private AppType _appType;
+		private BrokerType _brokerType;
+		protected bool isInitialized = false;
+		private readonly ILogger _logger = TestLogger.GetTestLogger();
+
+		public KafkaE2EFixture()
+		{
+			_logger.LogInformation($"Kafkae2efixture for {_language} {_appType} {_brokerType}");
+		}
+
+		public void OrchestrateInitialization()
+		{
```

Comment:
A suggestion:- I remember we are making sure that this cannot be contended bacause the current test suite infrastructure makes it impossible to call it in serial. In future you may change the way things execute, it is always good to protect the shared resources by the class which have it. Or there are known callers and would never(intended to ) change.

Reply:
Makes sense. We can come back to this later.